src/image/image_type.cpp

```diff
@@ -7,6 +7,19 @@
 
 namespace turi{
 
+image_type::image_type(boost::gil::rgb8_image_t gil_image) {
+  auto it = view(gil_image).begin();
```

Comment:
I really dislike the use of `auto` as a convenience for the writer. Code is read so much more often than it is written. I think usually its' usage doesn't make sense, there are of course exceptions.

You should probably get use rid of the usage of `auto` on this line and the next.

Reply:
Eh, this auto is hard with iterator types.

Reply:
Got rid of the auto!

Reply:
Ok, that sounds like an exception. I was mainly talking about the one that was on the next line.